test/sshkit/ssh/channel_functional_test.exs

```diff
@@ -0,0 +1,33 @@
+defmodule SSHKit.SSH.ChannelFunctionalTest do
+  @moduledoc false
+
+  use SSHKit.FunctionalCase, async: true
+
+  @bootconf [user: "me", password: "pass"]
+
+  describe "Channel.subsystem/3" do
+    @tag boot: [@bootconf]
+    test "with user", %{hosts: [host]} do
+      {:ok, conn} = SSHKit.SSH.connect(host.name, host.options)
+      {:ok, channel} = SSHKit.SSH.Channel.open(conn)
+      :success = SSHKit.SSH.Channel.subsystem(channel, "greeting-subsystem")
+
+      SSHKit.SSH.Channel.send(channel, "Lorem\n")
+
+      [welcome_message, response_message, _] = get_messages(channel)
+
+      assert welcome_message == "Hello, who am I talking to?"
+      assert response_message == "It's nice to meet you Lorem"
+    end
+  end
+
+  defp get_messages(channel, message \\ "") do
+    {:ok, {:data, _channel, _type, next_line}} = SSHKit.SSH.Channel.recv(channel)
+
+    if String.ends_with?(next_line, "Lorem\n") do
+      String.split("#{message}#{next_line}", "\n")
+    else
+      get_messages(channel, "#{message}#{next_line}")
+    end
+  end
```

Comment:
🤔 Indeed, the hard-coded `"Lorem\n"` in the `get_messages/2` function seems slightly counter-intuitive.

If we assume a simple text-/line-based protocol, how about something like the following:

```elixir
defp readlines(channel, count, buffer \\ "") do
  lines = String.split(buffer)
  
  if Enum.count(lines) < count do
    {:ok, {:data, ^channel, _, data}} = SSHKit.SSH.Channel.recv(channel)
    readlines(channel, count, buffer <> data)
  else
    Enum.take(lines, count)
  end
end
```
?

Then in the test we could write:

```elixir
[welcome_message, response_message] = readlines(channel, 2)
```

or, we could even make sure we receive the question first before we send our name/reply:

```elixir
assert ["Hello, who am I talking to?"] == readlines(channel, 1)

:ok = SSHKit.SSH.Channel.send(channel, "Lorem\n")

assert ["It's nice to meet you Lorem"] == readlines(channel, 1)
```

_Note: This is just a quick braindump, I didn't actually try to implement & run this._

Reply:
Specifying that we want to receive only one message would give a problem if we answer before waiting for messages, because we'd be at risk of receiving two messages.

But your suggestion of validating the first line before replying is great! This simplifies the test a lot because we'd only ever get one single message at a time.